src/ui/components/common/content_actions.rs

```diff
@@ -1,121 +1,135 @@
 use crate::{
   contexts::site_resource_context::SiteResource,
   serverfns::users::create_block_user_action,
-  ui::components::common::icon::{Icon, IconType},
+  ui::components::common::{
+    fedilink::Fedilink,
+    icon::{Icon, IconType},
+  },
   utils::{
     derive_user_is_logged_in,
-    types::{ServerAction, ServerActionFn},
+    traits::ToStr,
+    types::{ContentActionType, ServerAction, ServerActionFn},
   },
 };
+use comment_count::CommentCount;
+use hide_post_button::HidePostButton;
 use leptos::*;
 use leptos_router::{ActionForm, A};
+use report_button::ReportButton;
+use tailwind_fuse::tw_join;
 
-mod post_content_actions;
-pub use post_content_actions::PostContentActions;
-
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-enum ContentActionType {
-  Post {
-    comments: MaybeSignal<i64>,
-  },
-  #[allow(dead_code)]
-  Comment,
-}
+mod comment_count;
+mod hide_post_button;
+mod report_button;
 
 #[component]
-fn ContentActions<SA, RA>(
+pub fn ContentActions<SA>(
   content_action_type: ContentActionType,
-  #[prop(into)] id: MaybeSignal<i32>,
+  id: i32,
+  saved: Signal<bool>,
   save_action: ServerAction<SA>,
-  #[prop(into)] saved: MaybeSignal<bool>,
-  report_action: ServerAction<RA>,
-  #[prop(into)] creator_id: MaybeSignal<i32>,
+  creator_id: i32,
```

Comment:
I'll have to take the tuple value out of the `PersonId` at some point. Doing so once in the post listing removes the need to do it multiple times in the content actions. I tried getting the best of both worlds by making the `creator_id` an into prop, but sadly `PersonId` doesn't implement `Into<i32>`.

Reply:
I'd need to get into the code and start playing, but I thought signals were generic, and those `#[prop(into)` macros are optional. So you could do `ReadSignal<PostId>` 

The only place I can think of where you'd absolutely need to take the tuple out, is in route params, and then you should probably wrap it in `PostId(` or whatever type to make sure it uses that ID type internally.

Reply:
> I'd need to get into the code and start playing, but I thought signals were generic, and those #[prop(into) macros are optional. So you could do ReadSignal<PostId>

Since the id for a post, comment, or creator isn't something that can change (barring the report modal, since there's only one of them and it gets reused for each report), there's no need to make it reactive. In this case, a signal would just add extra overhead.

> The only place I can think of where you'd absolutely need to take the tuple out, is in route params, and then you should probably wrap it in PostId( or whatever type to make sure it uses that ID type internally.

It's also required to pass to the `value` attribute for hidden inputs in several places in the app.

At any rate, we'll need to take the tuple value out of the id at some point or another. As I stated above, doing so earlier rather than later reduces the number of times we need to do so. I can still change the components to use the `PostId`/`CommentId` if you still think it necessary in light of this information though.

Reply:
I do think its a good idea to keep those tuple types as much as possible. We could also probably add `Into<i32>` for those tuple structs in the lemmy codebase, but I'm not sure why that's necessary.

Reply:
Okay I opened up the code, and you should be fine using the `PersonId` or whichever internally, and then only extracting the inner when using it in html: ie  `value = {creator_id.0}`